### PR TITLE
Add Gumbel distribution

### PR DESCRIPTION
**Pull request by maxliving**
I'm pretty unfamiliar with this codebase, so it's possible I've done this incorrectly. Getting some errors in `test_variational_inference.py` (using Docker for testing, as described in `CONTRIBUTING.md`), but I think they're unrelated to this change.

### CODE REVIEW

File: pymc3/distributions/continuous.py [hunk 1]
```diff
@@ -1344,5 +1344,49 @@ def logp(self, value):
         return tt.switch(alltrue_elemwise([lower <= value, value < c]),
                          tt.log(2 * (value - lower) / ((upper - lower) * (c - lower))),
                          tt.switch(tt.eq(value, c), tt.log(2 / (upper - lower)),
-                         tt.switch(alltrue_elemwise([c < value, value <= upper]),
-                         tt.log(2 * (upper - value) / ((upper - lower) * (upper - c))),np.inf)))
+                                   tt.switch(alltrue_elemwise([c < value, value <= upper]),
```
Comment (twiecki) [inline pymc3/distributions/continuous.py]: these lines shouldn't change.

File: pymc3/distributions/continuous.py [hunk 2]
```diff
@@ -1344,5 +1344,49 @@ def logp(self, value):
...
+
+    def random(self, point=None, size=None, repeat=None):
+        mu, sd = draw_values([self.mu, self.beta], point=point)
+        return generate_samples(stats.gumbel_r.rvs, loc=mu, scale=sd,
+                                dist_shape=self.shape,
+                                size=size)
+
+    def logp(self, value):
+        scaled = (value - self.mu) / self.beta
+        return -scaled - tt.exp(-scaled) - tt.log(self.beta)
```
Comment (twiecki) [inline pymc3/distributions/continuous.py]: need `bound` for self.beta if it has only positive support.

**Comment by twiecki** [comment]
See minor comments, otherwise, this looks great!

**Comment by maxliving** [comment]
@twiecki Thanks for the quick review! Addressed your comments. Also do you have any thoughts on why the unit tests might be failing when run locally (in Docker) but not here?

File: pymc3/tests/test_distributions.py [hunk 3]
```diff
@@ -1,6 +1,8 @@
 from __future__ import division
 
 import itertools
+
+from pymc3.distributions.continuous import Gumbel
```
Comment (twiecki) [inline pymc3/tests/test_distributions.py]: this should move to the other imports

Comment (maxliving) [reply pymc3/tests/test_distributions.py]: Ahh, damn Pycharm. Done.

**Comment by maxliving** [comment]
@twiecki OK to merge?

**Comment by twiecki** [comment]
Thanks @maxliving!

**Pull request merged**